blockchain/chain/src/store/chain_store.rs

```diff
@@ -93,7 +94,7 @@ impl<DB> ChainStore<DB>
 where
     DB: Blockstore + Store + Send + Sync,
 {
-    pub fn new(db: DB, chain_config: Arc<ChainConfig>) -> Self
+    pub fn new(db: DB, chain_config: Arc<ChainConfig>, genesis_ts: Arc<Tipset>) -> Result<Self>
```

Comment:
It might be prettier to pass in a `BlockHeader` rather than `Arc<Tipset>`.

Reply:
Refactored this